src/averageAges.js

```diff
@@ -55,7 +71,29 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const filterWom = people.filter(wom => findMomByName(people, wom));
+
+  const kids = people.filter(kid =>
+    onlyWithSon
+      ? kid.sex === 'm'
+         && filterKidsByMom(people, kid) : filterKidsByMom(people, kid));
+
+  const ageDifference = kids.reduce((sum, kid) => {
+    const mother = filterWom.find(mom => mom.name === kid.mother);
+    const difference = kid.born - mother.born;
+
+    return sum + difference;
+  }, 0);
+
+  return ageDifference / kids.length;
+}
+
+function filterKidsByMom(people, kid) {
```

Comment:
Like line 34, the function name `filterKidsByMom` might be confusing. Consider renaming it to something like `hasMotherInPeople`.